ui-components/src/page_components/listing/listing_sidebar/Apply.tsx

```diff
@@ -66,11 +64,26 @@ const Apply = (props: ApplyProps) => {
     return method.type == ApplicationMethodType.FileDownload
   })
 
+  const getDropOffAddress = (addressType: ListingApplicationDropOffAddressType | undefined) => {
+    return addressType === ListingApplicationDropOffAddressType.leasingAgent
+      ? listing.leasingAgentAddress
+      : addressType === ListingApplicationDropOffAddressType.mailingAddress
+      ? listing.applicationMailingAddress
+      : listing.applicationDropOffAddress
+  }
+
+  const getPickUpAddress = (addressType: ListingApplicationPickUpAddressType | undefined) => {
+    return addressType === ListingApplicationPickUpAddressType.leasingAgent
+      ? listing.leasingAgentAddress
+      : addressType === ListingApplicationPickUpAddressType.mailingAddress
+      ? listing.applicationMailingAddress
+      : listing.applicationPickUpAddress
+  }
+
```

Comment:
Once you combine the address types, then you should be able to have one function for this that takes the address key whose value would be `applicationDropOffAddress` or `applicationPickUpAddress`